### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Shevtsooov**
[DEMO LINK](https://Shevtsooov.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +16,113 @@ export const goodsFromServer = [
...
+    }
+  });
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, getReversed] = useState(false);
```
Comment (dinaminahi) [inline src/App.tsx:55]: let's follow the naming convention for the useState. `const [value, setValue] = useState()`

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +16,113 @@ export const goodsFromServer = [
...
+  const [sortType, getSorted] = useState(SortType.NONE);
```
Comment (dinaminahi) [inline src/App.tsx:56]: same here

File: src/App.tsx [hunk 3]
```diff
@@ -15,48 +16,113 @@ export const goodsFromServer = [
...
+            className="
+              section__button
+              section__button--reset
+              is-danger
+              is-light
+            "
+            onClick={() => {
+              getSorted(SortType.NONE);
+              getReversed(false);
+            }}
```
Comment (dinaminahi) [inline src/App.tsx:115]: Create a function for this

**Review by dinaminahi — CHANGES_REQUESTED** [review]
Good job! 👍 
But few fixes needed

File: src/App.tsx [hunk 4]
```diff
@@ -15,48 +16,115 @@ export const goodsFromServer = [
...
+        {isSortingApplied && (
+          <button
+            type="button"
+            className="
+              section__button
+              section__button--reset
+              is-danger
+              is-light
+            "
+            onClick={() => reset()}
```
Comment (dinaminahi) [inline src/App.tsx:117]:
```suggestion
            onClick={reset}
```

File: src/App.tsx [hunk 5]
```diff
@@ -15,48 +16,115 @@ export const goodsFromServer = [
...
+    }
+  });
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReverse] = useState(false);
```
Comment (dinaminahi) [inline src/App.tsx:55]:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```

File: src/App.tsx [hunk 6]
```diff
@@ -15,48 +16,115 @@ export const goodsFromServer = [
...
+  const [sortType, setSort] = useState(SortType.NONE);
```
Comment (dinaminahi) [inline src/App.tsx:56]:
```suggestion
  const [sortType, setSortType] = useState(SortType.NONE);
```

**Review by dinaminahi — CHANGES_REQUESTED** [review]
You can name useState variables whatever you want, but it’s a convention for the second variable to use ‘set’ as a prefix with a camel case. `const [name, setName] = useState('Ihor')`

**Comment by Shevtsooov** [comment]
@dinaminahi Got you, corrected as well!

**Review by dinaminahi — APPROVED** [review]
Well done 👍

**Pull request closed without merging**